### PR TITLE
cody-gateway: report timeout error to client if deadline exceeded

### PR DESCRIPTION
**Pull request by bobheadxi**
Closes https://github.com/sourcegraph/sourcegraph/issues/54756

## Test plan

n/a

### CODE REVIEW

File: cmd/cody-gateway/internal/httpapi/embeddings/openai.go [hunk 1]
```diff
@@ -29,7 +27,7 @@ type openaiClient struct {
 	accessToken string
 }
 
-var tracer = otel.Tracer("cody-gateway/httpapi/embeddings")
```
Comment (bobheadxi) [inline cmd/cody-gateway/internal/httpapi/embeddings/openai.go:32]: unused

File: cmd/cody-gateway/internal/httpapi/embeddings/handler.go [hunk 2]
```diff
@@ -128,6 +128,14 @@ func NewHandler(
 					return
 				}
 
+				// More user-friendly message for timeouts
+				if errors.Is(err, context.DeadlineExceeded) {
```
Comment (eseliger) [inline cmd/cody-gateway/internal/httpapi/embeddings/handler.go:132]: are you sure this is the end where this error was happening, or was it the context on the sg side talking _to_ gateway? 
Regardless, this is probably a useful change.

**Review by eseliger — APPROVED** [review]

File: cmd/cody-gateway/internal/httpapi/embeddings/handler.go (hunk 2 above)
Comment (bobheadxi) [reply cmd/cody-gateway/internal/httpapi/embeddings/handler.go:132]: 😅 yeah, I guess we have to report that there too

Comment (bobheadxi) [reply cmd/cody-gateway/internal/httpapi/embeddings/handler.go:132]: I think 4d16618f79fd38a88c05d6307e68c448201b445f should be sufficient

Comment (bobheadxi) [reply cmd/cody-gateway/internal/httpapi/embeddings/handler.go:132]: expanding on the above - traces indicate the timeout is pretty much on our end, retrying upstream

**Pull request merged**